File: outputs/python/changed/DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1.py
# Copyright (c) 2019-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the license found in the
# LICENSE file in the root directory of this source tree.
#
import sys

def f_gold ( str1 ) :
    n = len ( str1 ) ;
    C = [ 0 ] * ( n + 1 ) ;
    P = [ [ False for x in range ( n + 1 ) ] for y in range ( n + 1 ) ] ;
    for i in range ( n ) :
        P [ i ] [ i ] = True ;
    for L in range ( 2 , n + 1 ) :
        for i in range ( n - L + 1 ) :
            j = i + L - 1 ;
            if ( L == 2 ) :
                P [ i ] [ j ] = ( str1 [ i ] == str1 [ j ] ) ;
            else :
                P [ i ] [ j ] = ( ( str1 [ i ] == str1 [ j ] ) and P [ i + 1 ] [ j - 1 ] ) ;
    for i in range ( n ) :
        if ( P [ 0 ] [ i ] == True ) :
            C [ i ] = 0 ;
        else :
            C [ i ] = sys.maxsize ;
            for j in range ( i ) :
                if ( P [ j + 1 ] [ i ] == True and 1 + C [ j ] < C [ i ] ) :
                    C [ i ] = 1 + C [ j ] ;
    return C [ n - 1 ] ;


#
def f_filled ( str ) :
    n = len ( str )
    C = [ [ ] for i in range ( n ) ]
    P = [ [ ] for i in range ( n ) ]
    i , j , k , L = 0 , 0 , 0 , 0
    for i in range ( n ) :
        P [ i ] [ i ] = True
    for L in range ( 2 , n + 1 ) :
        for i in range ( n - L + 1 ) :
            j = i + L - 1
            if L == 2 :
                P [ i ] [ j ] = ( str [ i ] == str [ j ] )
            else :
                P [ i ] [ j ] = ( str [ i ] == str [ j ] ) and P [ i + 1 ] [ j - 1 ]
    for i in range ( n ) :
        if P [ 0 ] [ i ] == True :
            C [ i ] = 0
        else :
            C [ i ] = int ( P [ i ] [ j ] )
            for j in range ( i ) :
                if P [ j + 1 ] [ i ] == True and 1 + C [ j ] < C [ i ] :
                    C [ i ] = 1 + C [ j ]
    return C [ n - 1 ]

if __name__ == '__main__':
    param = [
    ('YYGWgYrovdsh',),
    ('56109778',),
    ('101',),
    ('RxM',),
    ('187546405',),
    ('0110010',),
    ('wVODAmgvI',),
    ('56719',),
    ('10100011001100',),
    ('Wtpai',)
        ]
    n_success = 0
    for i, parameters_set in enumerate(param):
        if f_filled(*parameters_set) == f_gold(*parameters_set):
            n_success+=1
    print("%i, %i" % (n_success, len(param)))

File: outputs/java/changed/DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1.java
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

import java.util. *;
import java.util.stream.*;
import java.lang.*;

public class DYNAMIC_PROGRAMMING_SET_17_PALINDROME_PARTITIONING_1{
static int f_gold ( String str ) {
  int n = str . length ( ) ;
  int [ ] C = new int [ n ] ;
  boolean [ ] [ ] P = new boolean [ n ] [ n ] ;
  int i , j , k , L ;
  for ( i = 0 ;
  i < n ;
  i ++ ) {
    P [ i ] [ i ] = true ;
  }
  for ( L = 2 ;
  L <= n ;
  L ++ ) {
    for ( i = 0 ;
    i < n - L + 1 ;
    i ++ ) {
      j = i + L - 1 ;
      if ( L == 2 ) P [ i ] [ j ] = ( str . charAt ( i ) == str . charAt ( j ) ) ;
      else P [ i ] [ j ] = ( str . charAt ( i ) == str . charAt ( j ) ) && P [ i + 1 ] [ j - 1 ] ;
    }
  }
  for ( i = 0 ;
  i < n ;
  i ++ ) {
    if ( P [ 0 ] [ i ] == true ) C [ i ] = 0 ;
    else {
      C [ i ] = Integer . MAX_VALUE ;
      for ( j = 0 ;
      j < i ;
      j ++ ) {
        if ( P [ j + 1 ] [ i ] == true && 1 + C [ j ] < C [ i ] ) C [ i ] = 1 + C [ j ] ;
      }
    }
  }
  return C [ n - 1 ] ;
}


//
public static int f_filled ( String str1 ) {
    int n = str1 . length ( ) ;
    int [ ] C = {
    }
    P = new boolean [ n + 1 ] ;
    for ( int i = 0 ;  i < n ;  i ++ ) {
        P [ i ] [ i ] = true ;
    }
    for ( int L = 2 ;  L <= n ;  L ++ ) {
        for ( int i = 0 ;  i < n - L + 1 ;  i ++ ) {
            int j = i + L - 1 ;
            if ( ( L == 2 ) && ( str1 . length ( ) > 0 ) ) {
                P [ i ] [ j ] = ( str1 . charAt ( i ) == str1 . charAt ( j ) ) ;
            }
            else {
                P [ i ] [ j ] = ( ( str1 . charAt ( i ) == str1 . charAt ( j ) ) ? P [ i + 1 ] [ j - 1 ] : null ) ;
            }
        }
    }
    for ( int i = 0 ;  i < n ;  i ++ ) {
        if ( ( P [ 0 ] [ i ] == true ) && ( P [ 1 ] [ i ] == true ) ) {
            C [ i ++ ] = 0 ;
        }
        else {
            C [ i ] = Integer . MAX_VALUE ;
            for ( int j = 0 ;  j < i ;  j ++ ) {
                if ( ( P [ j + 1 ] [ i ] == true && 1 + C [ j ] < C [ i ] ) || ( P [ j + 1 ] [ i ] == false && 1 + C [ j ] < C [ i ] ) ) {
                    C [ i ] = 1 + C [ j ] ;
                }
            }
        }
    }
    return C [ n - 1 ] ;
}

public static void main(String args[]) {
    int n_success = 0;
    List<String> param0 = new ArrayList<>();
    param0.add("YYGWgYrovdsh");
    param0.add("56109778");
    param0.add("101");
    param0.add("RxM");
    param0.add("187546405");
    param0.add("0110010");
    param0.add("wVODAmgvI");
    param0.add("56719");
    param0.add("10100011001100");
    param0.add("Wtpai");
    for(int i = 0; i < param0.size(); ++i)
    {
        if(f_filled(param0.get(i)) == f_gold(param0.get(i)))
        {
            n_success+=1;
        }
    }
    System.out.println(n_success + " " + param0.size());
}
}